### PR TITLE
Add Identity Center Account Assignments to Unified Resource Cache

### PR DESCRIPTION
**Pull request by tcsc**
Adds Identity Center Account Assignments to the Unified resource cache so they can be requested in access requests.

Unfortunately we can't just include an `identitycenterv1.AccountAssignment` directly in the resource cache `ListResources` output because the legacy protobuf codegen used for the `authservice` and the new codegen used for `identitycenter/v1` produce incompatible serialization code, so resulting generated code does not compile.

To get around this issue, this change introduces a parallel (and slightly simplified) definition of an `IdentityCenterAccountAssignment` in the `authservice` protobuf spec to act as the wire format for this type. The cached `identitycenterv1.AccountAssignment` resources are copied into a  `proto.IdentityCenterAccountAssignment` on a cache read. 

I'm not particularly thrilled with this solution, and if anyone has an alternative that lets us use the definitions from `identitycenter/v1` without this parallel definition, I'd be much obliged.

Change Includes:
 - adding resources to cache
 - adding account assignment paginated resource
 - AccountAssignment Role Condition matching for RBAC

### CODE REVIEW

**Review by r0mant — COMMENTED** [review]
@tcsc Does this need a rebase? Feels like it contains code from PRs I already reviewed.

File: api/proto/teleport/legacy/client/proto/authservice.proto [hunk 1]
```diff
@@ -1934,6 +1934,42 @@ message CreateRegisterChallengeRequest {
...
+// IdentityCenterPermissionSet holds information about an Identity Center
+// permission set within an IdentityCenterAccountAssignment
+message IdentityCenterPermissionSet {
+  string ARN = 1;
+  string Name = 2;
+}
+
+// IdentityCenterAccountAssignment represents a requestable Identity Center
+// Account Assignment
+message IdentityCenterAccountAssignment {
```
[the remark this answers is not in the document]
Comment (tcsc) [reply api/proto/teleport/legacy/client/proto/authservice.proto:1955]: So, `types.IdentityCenterAccountAssignment` is for specifying Account Assignments in role conditions. If I had known bettrer back when defining this, I'd have called them an `AccountAssignmentCondition` or `AccountAssignmentExpression` (as they can be globbed).

This definition, `proto.IdentityCenterAccountAssignment` allows Account Assignment resources to be included in a list of resources returned by `ListResources` ( essentially by having a parallel (and slightly simplified) definition of an `identitycenterv1.AccountAssignment` in the legacy authservice protobuf definitions).

The primary `identitycenterv1.AccountAssignment` gets copied into a `proto.IdentityCenterAccountAssignment` at read time and then gets shipped over to whatever client invoked `ListResources().

The reason we can't just include an `identitycenterv1.AccountAssignment` in the `ListResources` output that the legacy protobuf codegen used for the `authservice` and the new codegen used for `identitycenter/v1` produce incompatible serialization code, so resulting generated code does not compile.

If there is a way around this that will let us use the definitions from `identitycenter/v1`, I'd be much happier.

**Comment by tcsc** [comment]
> @tcsc Does this need a rebase? Feels like it contains code from PRs I already reviewed.

And how.

File: api/client/client.go [hunk 2]
```diff
@@ -3822,6 +3826,26 @@ func (c *Client) ListResources(ctx context.Context, req proto.ListResourcesReque
 	}, nil
 }
 
+// repackIdentityCenterAccountAssignment converts a wire-format IdentityCenterAccountAssignment
+// resource back into an identitycenterv1.AccountAssignment instance.
+func repackIdentityCenterAccountAssignment(src *proto.IdentityCenterAccountAssignment) types.ResourceWithLabels {
```
Comment (r0mant) [inline api/client/client.go:3831]: Nit: Just a thought, can this maybe be moved to a more appropriate package where IC resources (or maybe `types`) and named `ICAccountAssignmentToResourceWithLabels`? We probably have other similar converters somewhere already, right? Like there's `types.LegacyTo153Metadata` you're using below.

Not a blocker though.

File: api/proto/teleport/legacy/client/proto/authservice.proto [hunk 3]
```diff
@@ -1940,6 +1940,50 @@ message CreateRegisterChallengeRequest {
   DeviceUsage DeviceUsage = 3 [(gogoproto.jsontag) = "device_usage,omitempty"];
 }
 
+// IdentityCenterAccount holds information about an Identity Center account
+// within an IdentityCenterAccountAssignment
+message IdentityCenterAccount {
+  string ID = 1;
```
Comment (r0mant) [inline api/proto/teleport/legacy/client/proto/authservice.proto:1946]: Nit: Godocs please on all exported fields, here and below.

**Review by r0mant — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Pull request merged**